### ISSUE TITLE
Not beeing able to add power-meter

### ISSUE THREAD

**Issue by JordVisbeen** [gruijter/com.gruijter.powerhour#16]
Hi,

When I try to add my power meter, I can only choose my PV-inverter, and it does not show my P1-smartmeter device (Electrical-part).

For P1-smartmeter device I use: 
https://homey.app/nl-nl/app/com.p1.dsmr/P1-DSMR-Board-Lezer/test/
https://github.com/BardoelT/com.p1.dsmr

Power by the Hour sees the P1-device, because adding the GAS-part of the meter is possibe (and working):
![image](https://user-images.githubusercontent.com/62200522/151348609-a8d0b78f-aedf-404f-9dd4-5c7ab54f44cc.png)

Any  idea on howto also add the Electrical-part of the P1-meter to Power by the Hour? 

Thanks for looking into it! 
Jordi

**Comment by gruijter** [comment]
Yes. The app com.p1.dsmr is not using the default capabilities: https://github.com/BardoelT/com.p1.dsmr/blob/ec64874fecabb25ba11b0d30df023002120cd110/drivers/p1-smartmeter/driver.compose.json#L29

I could maybe add these special capabilities to PBTH. Let me look into this.

**Comment by gruijter** [comment]
I have created a new version for you. Can you please test if it works?

https://homey.app/a/com.gruijter.powerhour/test

**Comment by JordVisbeen** [comment]
Wow that was quick! Thanks.

We're almost there, I'm now seeing a P1 meter, but it's old one with depricated driver (WS). I have that one still for testing the P1-DSMR purposes, but it's not connected to the meter on a daily bases.
![image](https://user-images.githubusercontent.com/62200522/151520708-02be801a-15c1-42c6-b129-7ba047d0692a.png)

I would like to see the API one, just what we have with the GAS-part (see picture in my first post).

Hope that can work too!

**Comment by gruijter** [comment]
The app has confusing drivers... I see that the current test release supports DSMR Logger, and DSMR Logger API V2. Now which one do you need to work?

**Comment by JordVisbeen** [comment]
Haha, well the only reason I'm using this is that the hardware costed me €7. I guess that comes with a price :)
Also the developer of the COM.P1.DSMR has changed in the past.

Since PBTH is 2 steps away from the actual firmware, it's hard to say how they connect. But my guess would be DSMR Logger API V2. 
The test release of com.p1.dsmr is especially made to support the API variant.

**Comment by gruijter** [comment]
Ok. I am working on a new test release for other features (money value, peak usage, etc). I will try to incorporate this driver addition into the next test release.

**Comment by JordVisbeen** [comment]
Hi,

Thanks for the work. What I see now is that I can select the P1 power meter (API), but the data is not coming through yet.
![image](https://user-images.githubusercontent.com/62200522/153035668-f20c4027-049a-4f0c-9d24-fe487506b54c.png)
![image](https://user-images.githubusercontent.com/62200522/153035783-918bb54f-8bfb-47e7-9ded-07a4a75c9851.png)

**Comment by gruijter** [comment]
In that case see this post for other options:

https://community.homey.app/t/app-pro-power-by-the-hour-insights-per-hour-day-month-and-year/20535/480

**Comment by JordVisbeen** [comment]
Thanks! I managed to get all desired meters via the virtual meters, and writing the data to them!
![image](https://user-images.githubusercontent.com/62200522/153233483-da478cf4-db13-4732-b6f1-8d3149a2355d.png)
Thanks for all your support!
